### ISSUE TITLE
Fail A11Y for hidden input field

### ISSUE THREAD

**Issue by phaldiya** [danialfarid/ng-file-upload#1176]
plugin creates a hidden file type input field that has no label associated with it.
that's why its failing A11Y - A form control does not have a corresponding label.

**Comment by phaldiya** [comment]
Solution is wrap <input type="file"> into <label> tag
like

<label><input type="file"></label>

I modified following function to achieve that

file --> ng-file-upload.js#655
```
 function createFileInput() {
      if (isInputTypeFile()) {
        return elem;
      }

      var fileElem = angular.element('<input type="file">');
      bindAttrToFileInput(fileElem);

      fileElem.css('visibility', 'hidden').css('position', 'absolute').css('overflow', 'hidden')
        .css('width', '0px').css('height', '0px').css('border', 'none')
        .css('margin', '0px').css('padding', '0px').attr('tabindex', '-1');
      generatedElems.push({el: elem, ref: fileElem});

      var fileLabel = angular.element('<label><span class="sr-only">attachment</span></label>');
      fileLabel.append(fileElem[0]);
      document.body.appendChild(fileLabel[0]);

      return fileElem;
 }
```

**Comment by danialfarid** [comment]
That is a hidden input so it shouldn't be checked for A11Y since it will never be selected for screen reader.
Would adding `aria-hidden=true` or hidden attribute to the input help. What validation tool are you using?

**Comment by phaldiya** [comment]
Hi Danial,

Thanks for quick replay.
No it won't work. I am using WAVE plugin for A11Y evaluation tool. Its
available for chrome and Firefox.

Let me know if you need more info.

--Pradeep


*Pradeep K. Haldiya*

IT Services | University of California
<http://www.linkedin.com/in/pradeephaldiya>
*P **Please consider the environment before printing this e-mail.*

**Comment by danialfarid** [comment]
You can submit a PR by changing `src/select.js`.
You won't need the `sr-only` span inside the label since it is a hidden input it will never be read by the screen reader.

**Comment by phaldiya** [comment]
i do, because if i don't put any text within the label it will fail A11Y - label without text.

**Comment by phaldiya** [comment]
not able to commit. need access rights

**Comment by danialfarid** [comment]
Fixed at 10.1.8

**Comment by phaldiya** [comment]
thanks
